frontend/src/metabase/plugins/index.ts

```diff
@@ -106,20 +108,37 @@ const AUTHORITY_LEVEL_REGULAR: CollectionAuthorityLevelConfig = {
   icon: "folder",
 };
 
+type AuthorityLevelMenuItem = {
+  title: string;
+  icon: string;
+  action: () => void;
+};
+
+type AuthorityLevelFormField = {
+  name: string;
+  title: string;
+  info?: string;
+  type: string;
+  options?: any[];
+};
+
 export const PLUGIN_COLLECTIONS = {
   AUTHORITY_LEVEL: {
     [JSON.stringify(AUTHORITY_LEVEL_REGULAR.type)]: AUTHORITY_LEVEL_REGULAR,
   },
   REGULAR_COLLECTION: AUTHORITY_LEVEL_REGULAR,
   isRegularCollection: (_: Collection | Bookmark) => true,
-  getAuthorityLevelFormFields: () => [],
+  getAuthorityLevelFormFields: (): AuthorityLevelFormField[] => [],
   getAuthorityLevelMenuItems: (
     _collection: Collection,
     _onUpdate: (collection: Collection, values: Partial<Collection>) => void,
-  ) => [],
+  ): AuthorityLevelMenuItem[] => [],
 };
 
-export const PLUGIN_COLLECTION_COMPONENTS = {
+export const PLUGIN_COLLECTION_COMPONENTS: Record<
+  string,
+  React.ComponentType<any>
```

Comment:
Can we avoid any here too? Maybe we could omit it or specify the props correctly